hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/federation/utils/FederationStateStoreFacade.java

```diff
@@ -1104,7 +1104,7 @@ public Collection<SubClusterInfo> getActiveSubClusters()
   public ApplicationSubmissionContext getApplicationSubmissionContext(ApplicationId appId) {
     try {
       GetApplicationHomeSubClusterResponse response = stateStore.getApplicationHomeSubCluster(
-          GetApplicationHomeSubClusterRequest.newInstance(appId));
+          GetApplicationHomeSubClusterRequest.newInstance(appId, true));
       ApplicationHomeSubCluster appHomeSubCluster = response.getApplicationHomeSubCluster();
```

Comment:
Why do we modify this parameter?

Reply:
I found that the originalSubmissionContext is null. And when call getApplicationSubmissionContext, GetApplicationHomeSubClusterRequest::setContainsAppSubmissionContext is not called, so will return null context.